media/public/include/IMediaPipeline.h

```diff
@@ -100,10 +100,15 @@ class IMediaPipeline
          * @param[in] type : The source type.
          * @param[in] mimeType : The mime type string.
          * @param[in] alignment : The alignment of media segment.
+         * @param[in] codecData : The additional data for decoder
+         * @param[in] streamFormat : The stream format
          */
         explicit MediaSource(int32_t id = 0, MediaSourceType type = MediaSourceType::UNKNOWN, const char *mimeType = "",
-                             SegmentAlignment alignment = SegmentAlignment::UNDEFINED)
-            : m_id(id), m_type(type), m_mimeType(mimeType), m_alignment(alignment)
+                             SegmentAlignment alignment = SegmentAlignment::UNDEFINED,
+                             const std::vector<uint8_t> &codecData = std::vector<uint8_t>(),
+                             StreamFormat streamFormat = StreamFormat::UNDEFINED)

```

Comment:
I'm being pedantic, but would it be better to have streamFormat before codecData. It would look neater and I suppose client more likely to set streamFormat than codecData?